### PR TITLE
Improve documentation and add Tox testing environment for it

### PR DESCRIPTION
**Pull request by tjanez**

### CODE REVIEW

File: resolwe/flow/models.py [hunk 1]
```diff
@@ -3,12 +3,24 @@
 Flow Models
 ===========
 
+Base Model
+**********
```
Comment (mstajdohar) [inline resolwe/flow/models.py]: It's strange to see stars *** a header level below equality ===. I prefer not to use stars at all:
```
===========
Flow Models
===========

Base Model
==========
```

File: resolwe/flow/models.py [hunk 2]
```diff
@@ -20,6 +32,43 @@
...
+Storage model
+*************
+
+Postgres ORM model for storing JSON.
+
+.. autoclass:: resolwe.flow.models.Storage
+    :members:
+
+
+Trigger model
```
Comment (mstajdohar) [inline resolwe/flow/models.py]: Remove triggers from docs and code. We do not use them and we do not envision using them in the future. Leftovers from the second generation.

File: resolwe/flow/models.py [hunk 3]
```diff
@@ -293,10 +368,12 @@ class Meta(BaseModel.Meta):
         (STATUS_DIRTY, 'Dirty')
     )
 
-    #: processor started date and time (set by :meth:`server.tasks.manager`)
+    #: processor started date and time (set by
```
Comment (mstajdohar) [inline resolwe/flow/models.py]: process not processor

process start date and time

File: resolwe/flow/models.py [hunk 4]
```diff
@@ -293,10 +368,12 @@ class Meta(BaseModel.Meta):
...
+    #: :meth:`resolwe.flow.executors.BaseFlowExecutor.run` or its derivatives)
     started = models.DateTimeField(blank=True, null=True)
 
-    #: processor finished date date and time (set by :meth:`server.tasks.manager`)
+    #: processor finished date date and time (set by
```
Comment (mstajdohar) [inline resolwe/flow/models.py]: process

File: resolwe/flow/utils/test.py [hunk 5]
```diff
@@ -307,60 +313,63 @@ def _assert_file(self, obj, fn_tested, fn_correct, compression=None, filter=lamb
...
+        :type obj: ~resolwe.flow.models.Data
 
-        :param str field_path: Path to list of file names in Data object.
+        :param str field_path: path to
+            :class:`~resolwe.flow.models.Data` object's field with the
+            list of file names
 
-        :param list fn_list: List od file names (and relative paths) of
-            files to which we want to compare. Name/path is relative to
+        :param list fn_list: list od file names (and relative paths) of
```
Comment (mstajdohar) [inline resolwe/flow/utils/test.py]: od -> of

File: resolwe/flow/utils/test.py [hunk 6]
```diff
@@ -307,60 +313,63 @@ def _assert_file(self, obj, fn_tested, fn_correct, compression=None, filter=lamb
...
+            files to compare against. Names/paths are relative to the
```
Comment (mstajdohar) [inline resolwe/flow/utils/test.py]: We could just write Paths instead of Names/paths.

Comment (mstajdohar) [inline resolwe/flow/utils/test.py]: Name is a path, relative to the directory that contains the file.

**Comment by mstajdohar** [comment]
Minor comments, please review.

File: resolwe/flow/models.py (hunk 1 above)
Comment (tjanez) [inline resolwe/flow/models.py]: Agreed.

File: resolwe/flow/models.py (hunk 2 above)
Comment (tjanez) [inline resolwe/flow/models.py]: Ok.

File: resolwe/flow/utils/test.py (hunk 6 above)
Comment (tjanez) [inline resolwe/flow/utils/test.py]: Agreed, I'll use "Paths are relative to the..."

**Comment by tjanez** [comment]
@mstajdohar, I've incorporated the fixes. Please re-review.

I'll remove the triggers in a separate pull request.

**Comment by mstajdohar** [comment]
LGTM

File: docs/conf.py [hunk 7]
```diff
@@ -1,13 +1,30 @@
...
-base_dir = os.path.join(os.path.abspath(os.path.dirname(__file__)), "..")
+import django
+from django.conf import settings
 
-os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'tests.settings')
+base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
 # Manual setup is required for standalone Django usage
+# NOTE: Since building documentation is independent from the tests, it can't
+# use the 'test.settings' Django settings module.
```
Comment (dblenkus) [inline docs/conf.py]: `test.settings` can be used to build documentation, but are not included in the build. Please update note to reflect that.

File: resolwe/flow/executors/__init__.py [hunk 8]
```diff
@@ -89,7 +97,7 @@ def run(self, data_id, script, verbosity=1):
 
         self.data_id = data_id
 
-        dir_mode = settings.FLOW_EXECUTOR.get('DATA_DIR_MODE', 0o755)
+        dir_mode = getattr(settings, 'FLOW_EXECUTOR["DATA_DIR_MODE"]', 0o755)
```
Comment (dblenkus) [inline resolwe/flow/executors/__init__.py]: In [1]: class A(object):
       ...:     pass
       ...:

    In [2]: a = A()

    In [3]: a.b = {'c': 1}

    In [4]: getattr(a, 'b["c"]', 0)
    Out[4]: 0

This is not working as expected.

File: resolwe/flow/executors/docker.py [hunk 9]
```diff
@@ -15,7 +15,7 @@ class FlowExecutor(LocalFlowExecutor):
 
     def __init__(self, *args, **kwargs):
         super(FlowExecutor, self).__init__(*args, **kwargs)
-        self.command = settings.FLOW_EXECUTOR.get('COMMAND', 'docker')
+        self.command = getattr(settings, 'FLOW_EXECUTOR["COMMAND"]', 'docker')
```
Comment (dblenkus) [inline resolwe/flow/executors/docker.py]: Same as above.

File: resolwe/flow/executors/local.py [hunk 10]
```diff
@@ -21,7 +21,7 @@ class FlowExecutor(BaseFlowExecutor):
 
     def __init__(self, *args, **kwargs):
         super(FlowExecutor, self).__init__(*args, **kwargs)
-        self.command = settings.FLOW_EXECUTOR.get('COMMAND', '/bin/bash')
+        self.command = getattr(settings, 'FLOW_EXECUTOR["COMMAND"]', '/bin/bash')
```
Comment (dblenkus) [inline resolwe/flow/executors/local.py]: And here.

File: resolwe/flow/executors/__init__.py (hunk 8 above)
Comment (tjanez) [inline resolwe/flow/executors/__init__.py]: @dblenkus, indeed, thanks for spotting the problem!

File: docs/conf.py (hunk 7 above)
Comment (tjanez) [inline docs/conf.py]: Done

**Comment by dblenkus** [comment]
LGTM

**Pull request merged**